lib/Basic/Default/TaskQueue.inc

```diff
@@ -143,12 +143,18 @@ bool TaskQueue::execute(TaskBeganCallback Began, TaskFinishedCallback Finished,
       stderrContents = stderrBuffer.get()->getBuffer();
     }
 
-    if (ReturnCode == -2) {
+#if defined(_WIN32)
+    // Wait() sets the upper two bits of the return code to indicate
+    // warnings (10) and errors (11).
+    if (ReturnCode & 0xC0000000) {
+#else
       // Wait() returning a return code of -2 indicates the process received
       // a signal during execution.
+    if (ReturnCode == -2) {
+#endif
```

Comment:
Nitpick: I dislike `#if`s that don't surround a whole unit of AST. Mind factoring this out into a boolean?

Reply:
Done!